lib/text/vtt_text_parser.js

```diff
@@ -843,6 +844,35 @@ shaka.text.VttTextParser = class {
 
     return (milliseconds / 1000) + seconds + (minutes * 60) + (hours * 3600);
   }
+
+  /**
+   * This method converts the HTML entities &amp;, &lt;, &gt;, &quot;, and
+   * &#39; in string to their corresponding characters.
+   *
+   * @param {!string} input
+   * @return {string}
+   * @private
+   */
+  static htmlUnescape_(input) {
+    /** Used to map HTML entities to characters. */
+    const htmlUnescapes = {
+      '&amp;': '&',
+      '&lt;': '<',
+      '&gt;': '>',
+      '&quot;': '"',
+      '&#39;': '\'',
+    };
+
+    /** Used to match HTML entities and HTML characters. */
+    const reEscapedHtml = /&(?:amp|lt|gt|quot|#(0+)?39);/g;
+    const reHasEscapedHtml = RegExp(reEscapedHtml.source);
+    if (input && reHasEscapedHtml.test(input)) {
```

Comment:
So you're calling `test` before you call `replace` because you want to avoid copying the string unnecessarily, right? Since this is going to be called on large strings, presumably.

I checked and `replace` looks to alway copy the string, even if nothing changes, so it makes sense as an optimization. But a comment explaining the reasoning might be good, so nobody tries to "optimize" this by removing the check.

Reply:
Done!